test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterOrChangeContactDetailsControllerSpec.scala

```diff
@@ -0,0 +1,315 @@
+          draftClaim.copy(mrnContactDetailsAnswer = Some(mrnContactDetailsAnswer))
+        val journeyStatus = FillingOutClaim(g, s, newClaim)
+        sessionData.copy(journeyStatus = Some(journeyStatus))
+      case _                                                         => fail()
+    }
+
+  def performAction(data: Seq[(String, String)] = Seq.empty)(action: Action[AnyContent]): Future[Result] =
+    action()(FakeRequest().withFormUrlEncodedBody(data: _*))
+
+  "Enter Or Change Contact Details controller" must {
+
+    "redirect to the start of the journey" when {
+
+      "there is no journey status in the session" in forAll(journeys) { journey =>
+        def performAction(): Future[Result] = controller.changeMrnContactDetails(journey)(FakeRequest())
+
+        val (session, _) = getSessionWithPreviousAnswer(None, None)
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session.copy(journeyStatus = None))
+        }
+
+        checkIsRedirect(
+          performAction(),
+          baseRoutes.StartController.start()
+        )
+
+      }
+
+    }
+
+    "display the page" when {
+
+      "the user has not answered this question before and is adding details" in forAll(journeys) { journey =>
+        def performAction(): Future[Result] = controller.enterMrnContactDetails(journey)(FakeRequest())
+
+        val session = getSessionWithPreviousAnswer(None, toSelectNumberOfClaims(journey).some)._1
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey("enter-or-change-contact-details.add.title")
+        )
+      }
+
+      "the user has not answered this question before and is changing details" in forAll(journeys) { journey =>
+        def performAction(): Future[Result] = controller.changeMrnContactDetails(journey)(FakeRequest())
+
+        val session = getSessionWithPreviousAnswer(None, toSelectNumberOfClaims(journey).some)._1
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey("enter-or-change-contact-details.change.title")
+        )
+      }
+    }
+
+    "handle submit requests" when {
+
+      " the user enters details" in forAll(journeys) { journey =>
+        val contactDetails = MrnContactDetails(nameData, emailData, phoneData)
+
+        val session        = getSessionWithPreviousAnswer(None, toSelectNumberOfClaims(journey).some)._1
+        val updatedSession = updateSession(session, contactDetails)
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+          mockStoreSession(updatedSession)(Right(()))
+        }
+
+        checkIsRedirect(
+          performAction(goodData.toSeq)(controller.changeMrnContactDetailsSubmit(journey)),
```

Comment:
Can you simplify this please, to something like:
`      def performAction(journeyBindable: JourneyBindable, data: (String, String)*): Future[Result] =
        controller.enterDuplicateMrnSubmit(journeyBindable)(FakeRequest().withFormUrlEncodedBody(data: _*))
`
and then 
`performAction(journey, data)`